modules/solid_mechanics/src/materials/EigenDecompMaterial.C

```diff
@@ -0,0 +1,65 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "EigenDecompMaterial.h"
+#include "RankTwoTensor.h"
+
+registerMooseObject("SolidMechanicsApp", EigenDecompMaterial);
+registerMooseObject("SolidMechanicsApp", ADEigenDecompMaterial);
+
+template <bool is_ad>
+InputParameters
+EigenDecompMaterialTempl<is_ad>::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addClassDescription(
+      "Creates material with eigenvalues and eigenvectors of a symmetric rank two tensor.");
```

Comment:
```suggestion
      "Emits material properties for the eigenvalues and eigenvectors of a symmetric rank two tensor.");
```

The fact that it has to be symmetric should be mentioned in the documentation. Also what happens if the tensor is not symmetric? You could just return the real part of the EVs then...

Reply:
It gives the wrong answer so I added an error check.